File: producer/api/models/question.js
const mongoose = require("mongoose");

const questionSchema = new mongoose.Schema({
    _id: mongoose.Schema.Types.ObjectId,
    form: {type: mongoose.Schema.Types.ObjectId, ref: "Form",required: true},
    text: {type: String}, 
    type: {type: String}
})

module.exports = mongoose.model("Question", questionSchema);

File: go-producer/models/question.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type QuestionSchema struct {
	Id   primitive.ObjectID `bson:"_id,omitempty"`
	Form primitive.ObjectID `bson:"form,omitempty" validate:"required"`
	Text string             `bson:"text,omitempty"`
	Type string             `bson:"type,omitempty"`
}
